### PR TITLE
Lazier images

### PR DESCRIPTION
**Pull request by ostowe**
This PR will:
* Apply only to the styled components version of the image component
* DRY out the image component a bit
* Load image src/srcset only on client side (much lazier than before)
* Use `IntersectionObserver` to only load images once onscreen

### CODE REVIEW

File: packages/styled/components/image/index.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  /**
+   * Image caption. This prop will modify the image markup to use a <figure>.
+   */
+  caption: PropTypes.string,
+  /**
+   * Additional classname(s) to add to image wrapper element
+   */
+  className: PropTypes.string,
+  /**
+   * Wether or not to lazyload this image via react-lazyload
```
Comment (pattiereaves) [inline packages/styled/components/image/index.js]:
```suggestion
   * Whether or not to lazyload this image via react-lazyload
```

File: packages/styled/components/image/index.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        {imageContent}
+      </Wrapper>
+      {(caption && showCaption) && <Caption>{caption}</Caption>}
+    </WrapperElement>
+  );
+};
+
+Image.propTypes = {
+  /**
+   * Alt text for the image
```
Comment (pattiereaves) [inline packages/styled/components/image/index.js]:
```suggestion
   * Alt text for the image.
```

File: packages/styled/components/image/index.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    PropTypes.number,
+    PropTypes.bool,
+    PropTypes.oneOf(['auto']),
+  ]).isRequired,
+  /**
+   * Image caption. This prop will modify the image markup to use a <figure>.
+   */
+  caption: PropTypes.string,
+  /**
+   * Additional classname(s) to add to image wrapper element
```
Comment (pattiereaves) [inline packages/styled/components/image/index.js]:
```suggestion
   * Additional classname(s) to add to image wrapper element.
```

File: packages/styled/components/image/index.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  /**
+   * Should this component render a `<picture>` element?
+   */
+  picture: PropTypes.bool,
+  /**
+   * Should the image caption be displayed?
+   */
+  showCaption: PropTypes.bool,
+  /**
+   * String with contents of <img> `sizes` attribute
```
Comment (pattiereaves) [inline packages/styled/components/image/index.js]:
```suggestion
   * String with contents of <img> `sizes` attribute
```
```suggestion
   * String with contents of <img> `sizes` attribute.
```

File: packages/styled/components/image/index.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+   */
+  sizes: PropTypes.string,
+  /**
+   * String with contents of <img> `src` attribute
```
Comment (pattiereaves) [inline packages/styled/components/image/index.js]:
```suggestion
   * String with contents of <img> `src` attribute.
```

File: packages/styled/components/image/index.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+   */
+  src: PropTypes.string.isRequired,
+  /**
+   * String with contents of <img> `srcset` attribute
```
Comment (pattiereaves) [inline packages/styled/components/image/index.js]:
```suggestion
   * String with contents of <img> `srcset` attribute.
```

File: packages/styled/components/image/index.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+   */
+  srcset: PropTypes.string.isRequired,
+  /**
+   * Array of objects containing values necessary for `<source>` tag attributes
```
Comment (pattiereaves) [inline packages/styled/components/image/index.js]:
```suggestion
   * Array of objects containing values necessary for `<source>` tag attributes.
```

File: packages/styled/components/image/index.js [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+   */
+  sourceTags: PropTypes.arrayOf(
+    PropTypes.shape({
+      /**
+       * String with contents of <source> `srcset` attribute
```
Comment (pattiereaves) [inline packages/styled/components/image/index.js]:
```suggestion
       * String with contents of <source> `srcset` attribute.
```

File: packages/styled/components/image/index.js [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+       */
+      srcset: PropTypes.string.isRequired,
+      /**
+       * Media query for which this `<source>` tag's srcset should apply
```
Comment (pattiereaves) [inline packages/styled/components/image/index.js]:
```suggestion
       * Media query for which this `<source>` tag's srcset should apply.
```

File: packages/styled/components/image/irvingImg.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+      onError={onError}
+      lazyload={lazyload}
+      ref={setNode}
+      hasSrc={hasSrc}
+    />
+  );
+};
+
+IrvingImg.propTypes = {
+  aspectRatio: PropTypes.oneOfType([
```
Comment (pattiereaves) [inline packages/styled/components/image/irvingImg.js]: The proptypes of `Image` are documented, should these be documented as well?

File: packages/styled/components/image/irvingPicture.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+        const { srcset: sourceSrcset, media } = source;
+        return (<source key={media} srcSet={sourceSrcset} media={media} />);
+      })}
+      {children}
+    </Picture>
+  );
+};
+
+IrvingPicture.propTypes = {
+  aspectRatio: PropTypes.oneOfType([
```
Comment (pattiereaves) [inline packages/styled/components/image/irvingPicture.js]: Should these propTypes be documented?

File: packages/styled/components/image/irvingImg.js (hunk 10 above)
Comment (ostowe) [reply packages/styled/components/image/irvingImg.js]: They're a subset of the same parent image component props, so figured I'd just leave them off for documentation.

File: packages/styled/components/image/irvingPicture.js (hunk 11 above)
Comment (ostowe) [reply packages/styled/components/image/irvingPicture.js]: Same as above, they're just a subset.

**Pull request merged**